rmw_zenoh_cpp/src/detail/attachment_helpers.hpp

```diff
@@ -15,34 +15,37 @@
 #ifndef DETAIL__ATTACHMENT_HELPERS_HPP_
 #define DETAIL__ATTACHMENT_HELPERS_HPP_
 
-#include <zenoh.h>
+#include <cstdint>
+#include <vector>
+
+#include <zenoh.hxx>
 
 #include "rmw/types.h"
 
 namespace rmw_zenoh_cpp
 {
-///=============================================================================
 class AttachmentData final
 {
 public:
-  AttachmentData(
-    const int64_t sequence_number,
-    const int64_t source_timestamp,
-    const uint8_t source_gid[RMW_GID_STORAGE_SIZE]);
-  explicit AttachmentData(const z_loaned_bytes_t *);
+  explicit AttachmentData(
+    const int64_t _sequence_number,
+    const int64_t _source_timestamp,
+    const std::vector<uint8_t> _source_gid);
+
+  explicit AttachmentData(const zenoh::Bytes & bytes);
```

Comment:
So that `bytes` can be moved into the Deserializer.
```suggestion
  explicit AttachmentData(zenoh::Bytes bytes);
```

Reply:
This is already done in the method
```cpp
  zenoh::ext::Deserializer deserializer(std::move(attachment));
````

Reply:
In point of fact, I think that this should be a constref (i.e. it is fine as-is), and further, we shouldn't have the `std::move` inside the method.  That's because inside of the method, what we are doing is taking the the set of bytes, and deserializing.  That's going to involve a copy anyway, so we don't want another copy when the call is made, and further, this method doesn't want to take ownership of the bytes.  So I think this is good as-is, and I'll add another comment for removing that `std::move`.